components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/DefaultUserStoreAttributeMappingParser.java

```
@@ -0,0 +1,160 @@
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_URI;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECT;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECTS;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DISPLAY_NAME;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.LOCAL_DIALECT_URL;
+
+/**
+ * Parser to read default user store mappings from claim-config.xml.
+ */
+public class DefaultUserStoreAttributeMappingParser {
+
+    private static Map<String, UserStoreAttributeDO> defaultAttributes;
+    private static final Log LOG = LogFactory.getLog(DefaultUserStoreAttributeMappingParser.class);
+
+    private DefaultUserStoreAttributeMappingParser() {
+
+        init();
+    }
+
+    private static final class ParserHolder {
+
+        static final DefaultUserStoreAttributeMappingParser PARSER = new DefaultUserStoreAttributeMappingParser();
+    }
+
+    public static DefaultUserStoreAttributeMappingParser getInstance() {
+
+        return ParserHolder.PARSER;
+    }
+
+    private static void init() {
+
+        File attributeMappingXml = new File(CarbonUtils.getCarbonConfigDirPath(), CLAIM_CONFIG);
+
+        if (attributeMappingXml.exists()) {
+            try (InputStream inStream = new FileInputStream(attributeMappingXml)) {
+                if (inStream == null) {
+                    String message = String.format("Claim-config configuration is not found at: %s/%s.",
+                            CarbonUtils.getCarbonConfigDirPath(), CLAIM_CONFIG);
+                    throw new FileNotFoundException(message);
+                }
+                buildDefaultAttributeMapping(inStream);
+            } catch (FileNotFoundException e) {
```

I am bit confused here.. Why would we need to throw a file not found exception from line 84 and catch it here ? Exception handling is a cost. We don't need to do it unless we really need it

---

Seems to be `inStream` is never null and what's caught from here is the exception thrown from the `FileInputStream` constructor.

---

Yes it is true. I'm fixing it.

---

fixed https://github.com/wso2/carbon-identity-framework/pull/3781